### ISSUE TITLE
When  'Enable default away message' is uset you cant set yourself as away.

### ISSUE THREAD

**Issue by un1versal** [kvirc/KVIrc#2101]
## Intro

if you disable away message you disable ability to set away. 
The away message and the ability for setting away should not be tied together.

you should be able to **set yourself as away** **FROM UI or command line** even if the message is unset `/option boolUseAwayMessage 0` and away message is empty.

![capture](https://cloud.githubusercontent.com/assets/3521959/17616389/54151982-606c-11e6-8741-a0c0cafe3ac3.PNG)
### Expected behavior

Being able to set yourself away even if no message is defined or enabled in UI
### Actual behavior

You can only set yourself away via command line if `/option boolUseAwayMessage 0`
### Steps to reproduce the issue
1. run /option boolUseAwayMessage 0
2. click either or the away button(s) on UI

toolbar
![capture](https://cloud.githubusercontent.com/assets/3521959/17617452/37d31bce-6072-11e6-85dc-cc3cf2ec9a41.PNG)
statusbar
![capture2](https://cloud.githubusercontent.com/assets/3521959/17617453/37d499e0-6072-11e6-961b-7e25f0a781d9.PNG)

Or tray icon but this is  broken atm. see https://github.com/kvirc/KVIrc/issues/2102
3. see the message `[08:45:54] [Leaving away status]: You are no longer marked as being away` but you are not actually away or returning from away.
### System information

```
KVIrc 4.9.2 'Aria'

Runtime Info:
System name: Linux 4.4.0-34-generic
System version: #53-Ubuntu SMP Wed Jul 27 16:06:39 UTC 2016
Architecture: x86_64
Qt version: 5.5.1
Qt theme: fusion

Build Info:
Build date: 2016-08-12 07:37:08 UTC
Sources date: 20160812
Revision number: git-6918-g5f84c46
System name: Linux-4.4.0-34-generic
CPU name: x86_64
Build command: /usr/bin/cmake
Build flags: 
   MANDIR=share/man
   CMAKE_INSTALL_PREFIX=/usr
   LIB_SUFFIX=/x86_64-linux-gnu
   Threads=POSIX
Compiler name: /usr/bin/c++
Compiler flags: N/A
Qt version: 5.5.1
Features: IRC, Debug, Crypt, SSL, IfAddr, IPC, OSS, Transparency, Webkit, Perl, Python, Enchant, Qt5, KVS
```

**Comment by un1versal** [comment]
also see https://github.com/kvirc/KVIrc/issues/1360#issuecomment-239404501

**Comment by staticfox** [comment]
Read the documentation on `/away` as specifieced by KviKvsCoreCommands

```
If no away reason is specified, the behaviour of this command depends from the value
of the option boolUseAwayMessage: if enabled, user will be set away using the default
away message (or [i]Away from keyboard[/i] if no default away message is set);
otherwise, the user will return from a previous away state.
```

That said, this is not a bug, more undesired behavior. Would you rather it changed?

**Comment by un1versal** [comment]
> That said, this is not a bug, more undesired behavior. 

lol, not a bug... OK then

> Would you rather it changed?

yes please, to something that makes sense. ie.

you should be able to set yourself as away FROM UI or command line even if  `/option boolUseAwayMessage 0` and away message is empty.

if the away message is empty then it should set the message added in patch https://github.com/kvirc/KVIrc/commit/f3613a7c74506d05bc670080a35bcc622e58daf9

so then it all makes sense at least and all ties in together nicely.

**Comment by AndrioCelos** [comment]
As @staticfox mentioned, `/away` without a parameter will lower your away flag if `boolUseAwayMessage` is `false`, which mirrors the IRC protocol `AWAY` command. This is intentional.

The problem is that [clicking the Away button on the toolbar](https://github.com/kvirc/KVIrc/blob/master/src/kvirc/kernel/KviCoreActions.cpp#L1056), or [double-clicking on the status bar widget](https://github.com/kvirc/KVIrc/blob/master/src/kvirc/ui/KviStatusBarApplet.cpp#L209), call parameterless `/away`, and hence will lower your away flag if the option is `false`.

**Comment by staticfox** [comment]
I'm fine with changing the UI I guess, but I don't think `/away` command line should be altered

**Comment by un1versal** [comment]
I **had** several problems with the way it **didnt work**, but main one, is that is that no matter how you look at it, it makes no sense to make this return from a previous away state, when you 

A) cant even set an away status when **/option boolUseAwayMessage is false** via UI  
B) youre not away to begin with, so cant very well return from a a state you cant set or even be.

the command line you cant even set an empty away either in this condition, it must be non empty which also makes default message useless.

as is now and after testing the proposed patch #2104, at least things work in some logical manner finally at long last and no kittens have been harmed in the process (cat #624) cat 0 vs spider 623 ;)

**Comment by staticfox** [comment]
`AWAY` without parameters means "Return from away status", that's literally how the command works. So it totally makes sense that `/away` returns you from away. That goes for both client to server and server to server protocol.

**Comment by un1versal** [comment]
that was never an issue in itself.

you literally also could not do /away "" it had to be a non blank message which completely went ignored the default message patched into fiixing  #1360

now you can do /away "" or /away -d
